src/Nowruz/modules/userProfile/containers/editInfoOrg/useEditInfoOrg.tsx

```diff
@@ -0,0 +1,225 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { debounce } from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useDispatch, useSelector } from 'react-redux';
+import { SOCIAL_CAUSES } from 'src/constants/SOCIAL_CAUSES';
+import { socialCausesToCategory } from 'src/core/adaptors';
+import { Location, Organization, User, getIndustries, identities, preRegister, searchLocation } from 'src/core/api';
+import { checkUsernameConditions } from 'src/core/utils';
+import store, { RootState } from 'src/store';
+import { setIdentityList } from 'src/store/reducers/identity.reducer';
+import { updateOrgProfile } from 'src/store/thunks/profile.thunks';
+import * as yup from 'yup';
+
+const schema = yup
+  .object()
+  .shape({
+    name: yup.string().required('Organization name is required'),
+    username: yup.string().required('Username is required'),
+    city: yup.object().shape({
+      label: yup.string().required('City is required'),
+      value: yup.string(),
+    }),
+    country: yup.string(),
+    industry: yup.object().shape({
+      label: yup.string().required('Industry is required'),
+      value: yup.string(),
+    }),
+    summary: yup.string().required('Summary is required'),
+    socialCauses: yup
+      .array()
+      .of(
+        yup.object().shape({
+          label: yup.string(),
+          value: yup.string(),
+        }),
+      )
+      .min(1, 'Select at least one social cause'),
+  })
+  .required();
+
+export const useEditInfoOrg = (handleClose: () => void) => {
+  const dispatch = useDispatch();
+  const org = useSelector<RootState, User | Organization | undefined>((state) => {
+    return state.profile.identity;
+  }) as Organization;
+
+  const [isUsernameValid, setIsusernameValid] = useState(false);
+  const [isUsernameAvailable, setIsusernameAvailable] = useState(false);
+  const [letterCount, setLetterCount] = useState(org.mission?.length);
+  const keyItems = Object.keys(SOCIAL_CAUSES);
+  const socialCauseItems = keyItems.map((i) => {
+    return { value: SOCIAL_CAUSES[i].value, label: SOCIAL_CAUSES[i].label };
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+    setError,
+    clearErrors,
+    watch,
+    setValue,
+    reset,
+  } = useForm({
+    mode: 'all',
+    resolver: yupResolver(schema),
+    defaultValues: {
+      name: org.name,
+      username: org.shortname,
+      city: { label: org.city, value: org.city },
+      country: org.country,
+      summary: org.mission,
+      industry: { label: org.industry, value: org.industry },
+      socialCauses: socialCausesToCategory(org?.social_causes),
+    },
+  });
+
+  const username = watch('username');
+  const checkUsernameAvailability = async (username: string) => {
+    const checkUsername = await preRegister({ username });
+    if (checkUsername.username === null) {
```

Comment:
```suggestion
      setIsusernameAvailable(checkUsername.username === null);
```

We can actually remove the condition